subprojects/common/http-client/src/main/kotlin/com/avito/http/HttpClientProvider.kt

```diff
@@ -1,24 +1,29 @@
 package com.avito.http
 
 import com.avito.android.stats.StatsDSender
+import com.avito.http.internal.RequestMetadataProvider
 import com.avito.http.internal.StatsHttpEventListener
+import com.avito.http.internal.TagRequestMetadataProvider
 import com.avito.logger.LoggerFactory
 import com.avito.time.TimeProvider
 import okhttp3.OkHttpClient
 
 public class HttpClientProvider(
     private val statsDSender: StatsDSender,
     private val timeProvider: TimeProvider,
-    private val loggerFactory: LoggerFactory
+    private val loggerFactory: LoggerFactory,
 ) {
 
     private val builder = OkHttpClient.Builder()
 
-    public fun provide(): OkHttpClient.Builder {
+    public fun provide(
+        requestMetadataProvider: RequestMetadataProvider = TagRequestMetadataProvider()
```

Comment:
I see the problem: we create an `OkHttpClient.Builder` with `RequestMetadataProvider` so it adds an implicit requirement that the request must contain a tag; on the opposite side we describe tag for requests in retrofit so I'm easily could forget to add a tag to requests or create Retrofit service with incorrect `OkHttpClient`

I propose unite Retrofit Services creation with adding `RequestMetadataProvider` for `OkHttpClient` used by those Retrofit services

Reply:
i don't quite understand your proposal; let's discuss it on next sync

p.s.
this module doesn't know about retrofit and also used with services which use okhttp directly (without retrofit)

Reply:
Ok